db/large_data_handler.cc

```diff
@@ -46,17 +46,22 @@ large_data_handler::large_data_handler(uint64_t partition_threshold_bytes, uint6
         partition_threshold_bytes, row_threshold_bytes, cell_threshold_bytes, rows_count_threshold);
 }
 
-future<bool> large_data_handler::maybe_record_large_partitions(const sstables::sstable& sst, const sstables::key& key, uint64_t partition_size) {
+future<std::pair<bool, bool>> large_data_handler::maybe_record_large_partitions(const sstables::sstable& sst, const sstables::key& key, uint64_t partition_size, uint64_t rows_count) {
     assert(running());
-    if (partition_size > _partition_threshold_bytes) {
+    std::pair<bool, bool> record{
+        __builtin_expect(partition_size > _partition_threshold_bytes, false),
```

Comment:
we don't need `__builtin_expect` here since there's no branching involved.

Reply:
d'oh :)